python/ray/autoscaler/_private/aws/config.py

```diff
@@ -107,89 +100,105 @@ def log_to_cli(config):
     cli_logger.doassert(provider_name is not None,
                         "Could not find a pretty name for the AWS provider.")
 
+    head_node_type = config["head_node_type"]
+    head_node_config = config["available_node_types"][head_node_type][
+        "node_config"]
+
     with cli_logger.group("{} config", provider_name):
```

Comment:
can we remove this?
Why do need to have a separation between head node type and available node types?

Reply:
IAM configuration is only set/logged for the head node type, and looks like we go out of our way to try to log details about the head node's type before the remaining types.

IAM configuration related stuff in the current version is here (head_node_config needed twice). IMO even if we only use the head_node_config once, it's still worth it to give it it's own variable since config["available_node_types"][config["head_node_type"]]["node_config"] is a bit painful

Reply:
shouldn't the IAM configuration be set to all the nodes?

Reply:
Wait, are you making sure to set it for the head node only? because the head node type might have max_workers>0 and hence you might accidentally set the IAM configs to these workers as well.

Reply:
Current discussion on that is [here](https://github.com/ray-project/ray/pull/15584#discussion_r624855269)